File: python/src/main/python/WaterIceCreamStore.py
from IceCreamStore import *
from StrawberryWaterIceCream import *
from PineappleWaterIceCream import *
from MangoWaterIceCream import *
from GuavaWaterIceCream import *


class WaterIceCreamStore(IceCreamStore):

    def createIceCream(self, flavor):
        if flavor == 'Strawberry':
            return StrawberryWaterIceCream()
        elif flavor == 'Pineapple':
            return PineappleWaterIceCream()
        elif flavor == 'Mango':
            return MangoWaterIceCream()
        elif flavor == 'Guava':
            return GuavaWaterIceCream()
        return None

File: csharp/factorypattern/main/stores/WaterIceCreamStore.cs
using factorypattern.main.icecreams;
using factorypattern.main.icecreams.water;

namespace factorypattern.main.stores
{
    internal class WaterIceCreamStore : IceCreamStore
    {

    override
     protected IceCream createIceCream(string flavor)
        {
            if (flavor.Equals("Strawberry"))
            {
                return new StrawberryWaterIceCream();
            }

            if (flavor.Equals("Pineapple"))
            {
                return new PineappleWaterIceCream();
            }
            if (flavor.Equals("Mango"))
            {
                return new MangoWaterIceCream();
            }
            return flavor.Equals("Guava") ? new GuavaWaterIceCream() : null;
        }
    }
}
